### ISSUE TITLE
Synpurge sometimes does not resolve room aliases

### ISSUE THREAD

**Issue by aperezdc** [aperezdc/synpurge#15]
Reported [by seequ in #matrix:matrix.org](https://matrix.to/#/!cURbafjkfsMDVwdRDQ:matrix.org/$1495544622373uiUFv:krtdex.com):

> Adrián 🎩: FTR, I haven't gotten synpurge working. It can't resolve room aliases and when given a room id, shouts

```
  File "/home/synapse/synpurge/lib/python3.4/site-packages/synpurge/purger.py", line 68, in get_purge_info
    return frozenset((info for room_id, info in self._rooms.items()))
TypeError: unhashable type: 'PurgeInfo'
```

**Comment by aperezdc** [comment]
Configuration file: https://gist.github.com/JaniM/49f0f81fae9213616c750492db6055f3

**Comment by aperezdc** [comment]
Changing line 20 in the configuration file to an alias, results in:

```
  File "/home/synapse/synpurge/lib/python3.4/site-packages/synpurge/purger.py", line 48, in __add
    self._rooms[room_id] = PurgeInfo(room_id, room_conf)
  File "<attrs generated init 48d5eec04f40c27481ec3cfcecab18c337901fab>", line 6, in __init__
  File "/home/synapse/synpurge/lib/python3.4/site-packages/attr/validators.py", line 32, in __call__
    attr, self.type, value,
TypeError: ("'room_id' must be <class 'str'> (got None that is a <class 'NoneType'>).", Attribute(name='room_id', default=NOTHING, validator=<instance_of validator for type <class 'str'>>, repr=True, cmp=True, hash=None, init=True, convert=None, metadata=mappingproxy({})), <class 'str'>, None)
```

**Comment by aperezdc** [comment]
[Also](https://matrix.to/#/!cURbafjkfsMDVwdRDQ:matrix.org/$1495545700383OneaR:krtdex.com):

> This is the pip version on python 3, on debian jessie, ftr

**Comment by neutralinsomniac** [comment]
Just chiming in to say that I have the same issue

**Comment by aperezdc** [comment]
This should be fixed now that PR #20 has been merged. Please update to the latest commit in the `master` branch, and feel free to re-open the issue if this surfaces again.

**Comment by JaniM** [comment]
```
synapse@vps340182:~/synpurge$ . bin/activate
(synpurge)synapse@vps340182:~/synpurge$ synpurge purge -d config.conf 
DEBUG:synpurge.cli:Using PostgreSQL: Database(pg=<postgresql.driver.pq3.Connection[pq://synapse_user:***@localhost:5432/synapse] b'I'>)
INFO:synpurge.cli:Resolving room aliases
```
That's all of the output with the latest commit.

Config:
```
[synpurge]
homeserver = https://krtdex.com
purge_request_timeout = 5 minutes
keep = 3 months
token = <snip>

[database]
user = synapse_user
password = <snip>
database = synapse
#port = 1234

clean_interval = 5
reindex_interval = 10

clean_full = true
reindex_full = true
```
